rfcs/NNNN-lit-3.0.md

```diff
@@ -0,0 +1,112 @@
+---
+Status: Active
+Champions: @justinfagnani
+PR: https://github.com/lit/rfcs/pull/16
+---
+
+# Lit 3.0
+
+A proposal for the next major release of Lit
+
+## Objective
+
+Allow us to make a few breaking changes that will let us reduce some engineering complexity, and very lightly reduce bundle size.
+
+### Goals
+- Require no code changes from developers using non-deprecated Lit 2.0 APIs
+- Drop IE support
+- Publish ES2021
+- Small cleanups (remove Lit pre-2.0 compatibility workarounds)
+
+### Non-Goals
+- Drop support for the Shady DOM polyfill
```

Comment:
Why not?

Reply:
It's still used even outside of IE.

Reply:
What's the compelling use case?

Reply:
Some of our users still force Shady DOM on, even in Chrome. It's a bad idea, but it's what they do. I don't think we should force them to use an older version yet.

Reply:
What would be the performance implications of dropping support?

Reply:
Basically nothing for external users. There are two separate Shady DOM affordances: 1) A couple of hooks for preparing templates and styles that are patched in the polyfill support files, 2) DOM operations in lit-html are wrapped in a function to support Shady DOM's no-patch mode. _But_ that wrapper function is only enabled with a specific build flag we use in Google. It's disabled on npm and the wrapper call is compiled away. So there's zero cost on npm.

And we test polyfills on modern browsers by forcing them on. Those tests aren't flaky. So we just don't gain much from removing support but we would have to maintain two versions internally if we did.

Reply:
@MrHBS you're allowed to think that.